lib/ash/query/function/ago.ex

```diff
@@ -3,28 +3,66 @@ defmodule Ash.Query.Function.Ago do
   Subtracts the given interval from the current time in UTC.
 
   For example:
-     deleted_at > ago(7, :days)
+     deleted_at > ago(7, :day)
 
   Documentation + available intervals inspired by the corresponding ecto interval implementation
   """
+
   use Ash.Query.Function, name: :ago
 
   def args, do: [[:integer, :duration_name]]
 
   def evaluate(%{arguments: [factor, interval]}) do
-    interval =
-      case interval do
-        :year -> :years
-        :month -> :months
-        :week -> :weeks
-        :day -> :days
-        :hour -> :hours
-        :minute -> :minutes
-        :second -> :seconds
-        :millisecond -> :milliseconds
-        :microsecond -> :microseconds
+    now = DateTime.utc_now()
+    shifted = datetime_add(now, -factor, interval)
+    {:known, shifted}
+  end
+
+  @doc false
+  def datetime_add(datetime, amount_to_add, unit)
```

Comment:
I added this as a public function to allow calling it from tests. Is it OK to keep it here? Or should it be in a helpers/utility module of some kind?